### DISCUSSION TITLE
HyperHDR First Time User

### DISCUSSION THREAD

**Discussion by joshlovell15** [awawa-dev/HyperHDR#285] [Q&A]
Hi Awawa and fellow companions,

As title states, I'm looking to move into the world of HyperHDR from Hyperion/WLED and can't quite wrap my head around it all. 

My current setup is Pi Zero W running Hyperion and two ESP's running WLED. Everything was working well until I decided to treat myself to a new OLED and AVR. As everyone does they go through all the settings and play with s**t till it breaks, well I have achieved that moment by playing with every 4K setting available and now unable to capture anything in Hyperion using the USB capture. (https://www.ebay.com.au/itm/164598782935)

Basically, I did this setup a few years ago and I'm well out of the loop for all of this HDR technology and would like a point in the right direction (PS I've read the blog and browsed the recent discussions on GH).

To my best knowledge, I need to find a capture card that supports 4K HDR10 input and USB3.0 capturing - is this correct? 
Would something like this work well or is it total overkill, I do not need any passthrough capabilities as I will be using the second output on my AVR to output video directly to the capture card. 
(https://www.elgato.com/en/game-capture-hd60-s-plus)

I am assuming I'd need to change the Pi Zero W out to a Pi 4B as I imagine the HDR processing and whatever magic Awawa has done requires a bit more oomph. 

I also will be storing the Pi in the TV unit so no need for the WLED interfaces anymore, can I cable directly into the GPIO of the Pi and remove the strain of WiFI communications to decrease output lag?

I plan on reusing my SK6812 RGBW strips if they are compatible but also have WS2815's on hand if need be. (I think from memory there is about 5.6M of LED's using 60LED/M SK6812)

All and any help with the hardware requirements for me to get onboard this project would be amazing! To anyone who does answer, thank you for your time in responding and assisting me.

Looking forward to playing with HyperHDR <3

Lounge Room Devices - 
LG 83" G2 OLED
DENON AVC-X6700H
APPLE TV 4K
XBOX ONE X
SONY PS5

**Comment by awawa-dev** [comment]
Hi

>  now unable to capture anything in Hyperion using the USB capture. https://www.ebay.com.au/itm/164598782935

This grabber (first ms2109 model, real USB2.0 not fake "USB3.0" with some bogus modifications) offers currently the best HDR to SDR quality [HDR2SDR QUALITY TEST](https://www.hyperhdr.eu/2022/04/usb-grabbers-hdr-to-sdr-quality-test.html) Not sure why it may not work: broken, try to capture 4k60 source and it's limited to 4k30, problems with cables or the splitter?

> To my best knowledge, I need to find a capture card that supports 4K HDR10 input and USB3.0 capturing - is this correct?  

Probably it can capture HDR10 but I'm not sure if it requires Windows & proprietary codec and that makes the grabber incompatible with HyperHDR. I have never heard of the success of using Elgato HD60S+ grabber with HyperHDR or similar application.

> Pi Zero W  

it's barely minimum. Will work with a grabber that has a scaler for one HyperHDR instance.
Safe choice: Rpi 3 or Rpi Zero 2 W for USB2.0 grabbers, Rpi4 for USB3.0 grabbers. Or you can try your luck with amlogic S905X3 tvbox from the latest review as Rpi4 substitute.

> I also will be storing the Pi in the TV unit so no need for the WLED interfaces anymore, can I cable directly into the GPIO of the Pi and remove the strain of WiFI communications to decrease output lag?

You can. HyperHDR allows that. But  I'm not recommend it or support such solutions.

> I plan on reusing my SK6812 RGBW strips if they are compatible but also have WS2815's on hand if need be. (I think from memory there is about 5.6M of LED's using 60LED/M SK6812)

SK6812 RGBW is currently best choice for LED strip. For such number of LEDs (~300-350) HyperSPI or HyperSerialESP32 is highly recommended. There is no guarantee if ws281x library will handle such number of LEDs directly from Rpi, locking system resources all the time to emulate SK6812 protocol (still not recommended and not supported).

**Comment by joshlovell15** [comment]
HI Awawa,

Thanks for the thorough and quick response!

Last night I got the image up and running on the Pi Zero and loaded up HyperHDR (UI feels very light and responsive compared to the Hyperion build I was running, not sure if you've lightened the CPU usage somehow but felt very quick!). I was facing some issues with the MS2109 I own trying to display live video in HyperHDR Live Video preview pane, pulled it out and threw it into OBS and it displays 4K30 & 1080p60 no worries. Put the USB Capture back into the Pi and rebooted to be sure, displays a frozen video feed in Live Video preview pane then nothing more, just remains frozen. 

So upon this and taking into account your responses and blog posts, I have decided to spend some more money to do it all the right way. 

I have concluded from your findings that the EZcap320 is the best solution whilst minimising the amount of devices needed. So with that I have begun looking for one to purchase, which is where I became stumped again. I was searching the EZcap official store on AliExpress and discovered that there is a 320, 320b & 326 that all have identical specifications. Is there a specific 320 series I should be looking for? links below :)

I also purchased a RPi4 8GB to be safe it wasn't an issue with the Pi I had (although it seemed healthy I'm removing all options from the equation)

I am purchasing some new Wemos D1 MINI's to run HyperSerialESP as well while I'm at it.

In doing all this I am under the assumption that the 4K60 HDR/Dolby Vision feed that comes from the AVR Monitor 2 output will go straight into the EZcap32XX of choice and then be downscaled and captured via USB3.0 into HyperHDR, from HyperHDR it can either be sent to an ESP32 running HyperSerialESP using USB or an ESP32 running HyperSerialWLED using WiFI (I understand there may be some delays using WLED due to background processes but have grown accustomed to the WLED suite so may choose this over USB)

I understand that this will not accept the XBOX or PS5 and for that to happen I would need to go from 4K120 HDR to 4K60 HDR or lower for the EZcap to recognize the resolution/format. This is where I would need a device like the HD Fury DR HDMI 8k if I'm correct?
https://hdfury.com/product/dr-hdmi-8k/

EZcap Model Links:
EZcap326 - https://www.aliexpress.com/item/1005003832309424.html
EZcap320B - https://www.aliexpress.com/item/1005003472092895.html
EZcap320 - https://www.aliexpress.com/item/1005001899008181.html

I know there's a lot in this post and I'm thankful of the time you've taken out of your day to assist!

Regards, J

Also congratulations on your 6th sponsor!

**Comment by awawa-dev** [comment]
Hi
> Last night I got the image up and running on the Pi Zero and loaded up HyperHDR (UI feels very light and responsive compared to the Hyperion build I was running, not sure if you've lightened the CPU usage somehow but felt very quick!). I was facing some issues with the MS2109 I own trying to display live video in HyperHDR Live Video preview pane, pulled it out and threw it into OBS and it displays 4K30 & 1080p60 no worries. Put the USB Capture back into the Pi and rebooted to be sure, displays a frozen video feed in Live Video preview pane then nothing more, just remains frozen. 

I would need HyperHDR logs taken 2-3 minutes after the application start to diagnose the possible problem with Rpi and your grabber. Please use latest 18.0.0.0beta3 version that is available for download. Maybe it's just USB power supply issue.

> I have concluded from your findings that the EZcap320 is the best solution whilst minimising the amount of devices needed. So with that I have begun looking for one to purchase, which is where I became stumped again. I was searching the EZcap official store on AliExpress and discovered that there is a 320, 320b & 326 that all have identical specifications. Is there a specific 320 series I should be looking for? links below :)

With latest dedicated LUT tables the situation has changed a bit. Still it's best all-in-one rather problematic free solution, but in terms of HDR to SDR quality there are currently better ones.

> I also purchased a RPi4 8GB to be safe it wasn't an issue with the Pi I had (although it seemed healthy I'm removing all options from the equation)

Yes, RPI4 is best choice though not the cheapest one and it's hard to find it in some countries.

> I am purchasing some new Wemos D1 MINI's to run HyperSerialESP as well while I'm at it.

I would recommend ESP32 rather... there are a huge number of reports of the counterfeit Wemos D1 MINI. Unless you are sure you will receive a unit with genuine CH340 the maybe search for ESP32 with CH9102X serial chip?

> In doing all this I am under the assumption that the 4K60 HDR/Dolby Vision feed that comes from the AVR Monitor 2 output will go straight into the EZcap32XX of choice and then be downscaled and captured via USB3.0 into HyperHDR, from HyperHDR it can either be sent to an ESP32 running HyperSerialESP using USB or an ESP32 running HyperSerialWLED using WiFI (I understand there may be some delays using WLED due to background processes but have grown accustomed to the WLED suite so may choose this over USB)

OK, but Dolby Vision is not supported by any video grabber I know. It's closed paid & proprietary technology. Only solutions for DV are HDFure Diva or jailbreaking webOS ( it's only for rooted LG TV but also solves 4k120 and VRR).

> I understand that this will not accept the XBOX or PS5 and for that to happen I would need to go from 4K120 HDR to 4K60 HDR or lower for the EZcap to recognize the resolution/format. This is where I would need a device like the HD Fury DR HDMI 8k if I'm correct?
https://hdfury.com/product/dr-hdmi-8k/

It has no scaler. Instead you need e.g. Vertex2 to change 4k120HDR to 1080p120HDR which Ezcap supports. But I didn't test that model and HDFurry's marketing leaves a lot of room for understatement, so I would first make sure it is capable of such a function with their support.

> EZcap Model Links:
EZcap326 - https://www.aliexpress.com/item/1005003832309424.html
EZcap320B - https://www.aliexpress.com/item/1005003472092895.html
EZcap320 - https://www.aliexpress.com/item/1005001899008181.html

I have & tested only Ezcap 320. There were of a lot of firmware issues, but most of them have been fixed. Still the quality of the USB-C connector is crap and sometimes it only recognizes USB2.0 when reconnected. Keep in mind it doesn't support HDCP. As an alternative FeinTech VSP01201/ (or Vertex2 for 4k120? _EDIT: if AVR is not capable of downscalling 4k120 to 1080_ ) + Ezcap 321 or MS2109 you already have.

> Also congratulations on your 6th sponsor!  

Thank you for your support :)
